fix(prompt-editor): show error warning for destructive env and conv var

Minamiyama opened the pull request:
## Summary

Add validation for environment and conversation variables to ensure they exist before rendering. Also check if node exists for non-environment/conversation variables. Invalid variables now show error styling consistently.

## Screenshots

| Before | After |
|--------|-------|
| ![84de3dd569a31f44c5e89f9bd8ac0d0](https://github.com/user-attachments/assets/c25dafcf-1f0f-4116-895d-90a9d419ecde) | ![802f7858b724b7267c9d66e4d8d5ea7](https://github.com/user-attachments/assets/68b64516-611f-405e-bf10-57945e097670) |

## Checklist

- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] I understand that this PR may be closed in case there was no previous discussion or issues. (This doesn't apply to typos!)
- [x] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [x] I've updated the documentation accordingly.
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

Minamiyama committed:
fix(prompt-editor): validate workflow variables before rendering

Add validation for environment and conversation variables to ensure they exist before rendering. Also check if node exists for non-environment/conversation variables. Invalid variables now show error styling consistently.

crazywoola reviewed (changes requested):
`useStore` only available in Chatflow/Workflow, it will cause errors in Chatbot and Agent. I suggest using props instead.

Minamiyama committed:
feat(workflow-variable-block): add support for environment and conversation variables

Pass environment and conversation variables directly to workflow variable blocks instead of using store. Update variable validation to check full path format.

Minamiyama committed:
refactor(prompt-editor): remove debug console logs from variable validation

Clean up code by removing unnecessary console.log statements used for debugging variable validation in workflow variable block component.

Minamiyama committed:
Merge branch 'langgenius:main' into bug/ui/show-error-info-for-missing-env-and-conv-var-in-prompt-editor

Minamiyama committed:
Merge branch 'langgenius:main' into bug/ui/show-error-info-for-missing-env-and-conv-var-in-prompt-editor

crazywoola reviewed (changes requested):
Add those env/conv variables remains red until switch to another node.

Minamiyama committed:
Merge branch 'langgenius:main' into bug/ui/show-error-info-for-missing-env-and-conv-var-in-prompt-editor

Minamiyama committed:
fix(workflow-variable-block): handle undefined environment and conversation variables

Add null checks for environmentVariables and conversationVariables in variable validation logic to prevent runtime errors when these props are undefined

Minamiyama commented:
> Add those env/conv variables remains red until switch to another node.

fixed

Minamiyama committed:
style(prompt-editor): normalize line endings and indentation in workflow variable block components
